BAZProjectC1/Components/CarosuelMenu/CarosuelMenuDelegate.swift

```diff
@@ -0,0 +1,12 @@
+//
+//  CarosuelMenuDelegate.swift
+//  BAZProjectC1
+//
+//  Created by 1044336 on 23/09/22.
+//
+import Foundation
+
+public protocol CarosuelMenuDelegate: AnyObject {
+    
+    func selectedOption(index: IndexPath)
+}
```

Comment:
Recuerda[ la convención de funciones para los delegates](https://github.com/raywenderlich/swift-style-guide#delegates). Podrías renombrar la función así:

```suggestion
public protocol CarouselMenuDelegate: AnyObject {
    func carouselMenu(_ carouselMenu: CarouselMenu, didSelectOption option: IndexPath)
}
```